### ISSUE TITLE
libsecp256k1 binary issue with mac M1

### ISSUE THREAD

**Issue by thunderbiscuit** [cashubtc/cashu#197]
Hi there. I'm just opening this for discoverability and to help others who might run into this issue!

Attempting to use the cashu package on a mac M1 will return this error (because the libsecp256k1 library only provides an `.so` file incompatible with the `x86_64` architecture):
```
ImportError: dlopen(/opt/homebrew/lib/python3.10/site-packages/secp256k1/_libsecp256k1.cpython-310-darwin.so, 0x0002): tried: '/opt/homebrew/lib/python3.10/site-packages/secp256k1/_libsecp256k1.cpython-310-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/usr/local/lib/_libsecp256k1.cpython-310-darwin.so' (no such file), '/usr/lib/_libsecp256k1.cpython-310-darwin.so' (no such file)
```

To successfully use cashu on the M1, one needs to run an `x86_64` version of Python. This is most easily done using Rosetta 2 and a collection of steps described in these posts:
 1. [Complex](https://towardsdatascience.com/how-to-use-manage-multiple-python-versions-on-an-apple-silicon-m1-mac-d69ee6ed0250), (didn't quite work for me)
 2. [Simpler but less robust](https://stackoverflow.com/a/71873666)

In short, use Rosetta to install an x86_64 version of homebrew. Use that homebrew to install an x86_64 version of Python. Create a virtual environment (or not) and pip install cashu there. Run cashu using the `arch -x86_64` Rosetta command (`arch -x86_64 cashu info`).
